src/SamplesApp/SamplesApp.UITests/Windows_UI_Xaml_Controls/PivotTests/UnoSamples_Pivot_Tests.cs

```diff
@@ -49,6 +49,23 @@ public void Pivot_Non_PivotItemChange_Validation()
 			_app.WaitForElement(_app.Marked("selectedItemTitle"));
 			_app.WaitForDependencyPropertyValue(selectedItemTitle, "Text", "item 1");
 			_app.WaitForDependencyPropertyValue(selectedItemContent, "Text", "My Item 1 Content");
-		}		
+		}
+
+		[Test]
+		[AutoRetry]
+		public void Pivot_Header_Affects_Render()
+		{
+			Run("UITests.Shared.Windows_UI_Xaml_Controls.Pivot.HeaderAffectsRender");
+
+			var pivot = _app.Marked("myPivot");
+			_app.WaitForElement(pivot);
+
+			// Get header text??
+
+			var btn = _app.Marked("changeHeaderBtn");
+			btn.FastTap();
+
+			// Get new header text??
```

Comment:
Any help on how to this? I remember trying something like `.Descendant("ImplicitTextBlock").GetText()` but it didn't work.

Reply:
You can probably search by using `.Class("")`

Reply:
@jeromelaban I've tried hard but couldn't really get something to work. Would love if you or someone can take a look

Reply:
@jeromelaban The problem with the complex header approach is that we set the header directly to a string in codebehind. So this doesn't change the TextBlock.

Changing the text of the TextBlock in codebehind will pass the test, but it's an "already working" scenario. So it doesn't test the fix and won't prevent regressions from happening.

Reply:
Where do you see the text being set from text?

It's supposed to be done from here: 

https://github.com/unoplatform/uno/blob/8d3724833105f78a3786b34b818b1c9cbc09a619/src/Uno.UI/UI/Xaml/Controls/Pivot/Pivot.cs#L168-L169

Reply:
@jeromelaban My last statement was about "Changing the text of the **TextBlock**". I supposed it shouldn't get this code path since we didn't really change the header property (I don't recall if I confirmed that at the time of writing that comment).

I'll try to write a runtime test for this anyway. Probably it's more easier and straightforward.